root/etc/e-smith/events/actions/nethserver-mattermost-conf

```diff
@@ -20,21 +20,54 @@
 # along with NethServer.  If not, see COPYING.
 #
 
-su - postgres -c "scl enable rh-postgresql94 -- psql --port=55432 -lqt | cut -d \| -f 1 | grep -q -w mattermost"
+su - postgres -c "scl enable rh-postgresql12 -- psql --port=55434 -lqt | cut -d \| -f 1 | grep -q -w mattermost"
 if [ $? -eq 1 ]; then # database do not exists
+    su - postgres -c "scl enable rh-postgresql94 -- psql --port=55432 -lqt | cut -d \| -f 1 | grep -q -w mattermost"
+    if [ $? -eq 0 ]; then
+        echo "Migrating data from postgresql 9.4 to postgresql 12..."
+        systemctl stop rh-postgresql94-postgresql
+        systemctl stop rh-postgresql12-postgresql
+        su - postgres -c "export LD_LIBRARY_PATH=/opt/rh/rh-postgresql94/root/usr/lib64:/opt/rh/rh-postgresql12/root/usr/lib64:$LD_LIBRARY_PATH && scl enable rh-postgresql12 -- pg_upgrade -b /opt/rh/rh-postgresql94/root/usr/bin -B /opt/rh/rh-postgresql12/root/usr/bin -d /var/opt/rh/rh-postgresql94/lib/pgsql/data -D /var/opt/rh/rh-postgresql12/lib/pgsql/data"
+        if [ $? -eq 0 ]; then
+            echo "Migration completed successfully"
+            systemctl start rh-postgresql94-postgresql
+            databases=$(su - postgres -c "scl enable rh-postgresql94 -- psql -q -A -t --port=55432 -c $'SELECT datname FROM pg_database WHERE datistemplate = false AND datname <> \'postgres\';'")
 
-    # create db and users
-    tmp_sql=`mktemp`
-    chmod a+r $tmp_sql
-    password=`perl -e "use NethServer::Password; print NethServer::Password::store('mattermost');"`
+            if [ $? -eq 0 ]; then
+                if [[ "$databases" == "mattermost" ]]; then
+                    # only mattermost database is present, delete all postgres 9.4 data
+                    systemctl stop rh-postgresql94-postgresql
+                    su - postgres -c "/var/lib/pgsql/delete_old_cluster.sh"
+                    /sbin/e-smith/config setprop rh-postgresql94-postgresql status disabled
+                    systemctl disable rh-postgresql94-postgresql
+                    rm -rf /usr/lib/systemd/system/rh-postgresql94-postgresql.service
+                    rm -rf /usr/lib/systemd/system/rh-postgresql94-postgresql@.service
```

Comment:
This should not be needed, I even fear those files are part of the postgres package and should not be removed